android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/Utils.kt

```diff
@@ -111,43 +107,3 @@ internal suspend fun <T> withRetries(logger: Logger, retryCount: Int, block: sus
         Result.Success(res)
     }
 }
-
-/**
- * A helper function which allows retrying a [block] of suspend code for a few times in case it fails.
- * Short-circuits execution if [block] returns [ServiceResult.AuthError] during any of its attempts.
- *
- * @param logger [Logger] that will be used to log retry attempts/results
- * @param retryCount How many retry attempts are allowed
- * @param block A suspend function to execute
- * @return A [ServiceResult] representing result of [block] execution.
- */
-internal suspend fun withServiceRetries(
-    logger: Logger,
-    retryCount: Int,
-    block: suspend () -> ServiceResult,
-): ServiceResult {
-    var attempt = 0
-    do {
-        attempt += 1
-        logger.info("withServiceRetries: attempt $attempt/$retryCount")
-        when (val res = block()) {
-            ServiceResult.Ok, ServiceResult.AuthError -> return res
-            ServiceResult.OtherError -> {}
-        }
-    } while (attempt < retryCount)
-
-    logger.warn("withServiceRetries: all attempts failed")
-    return ServiceResult.OtherError
-}
-
-internal suspend fun String?.asAuthFlowUrl(
```

Comment:
I believe we should avoid general refactoring in the PR that is aiming to refactor a specific part of the logic (decoupling), it makes reviewing much harder. How about we do it as a follow up and make it focused specifically on the code style?